src/app/clusters/basic-information/basic-information.cpp

```diff
@@ -163,7 +163,7 @@ CHIP_ERROR BasicAttrAccess::Read(const ConcreteReadAttributePath & aPath, Attrib
     case SoftwareVersionString::Id: {
         constexpr size_t kMaxLen                = DeviceLayer::ConfigurationManager::kMaxSoftwareVersionStringLength;
         char softwareVersionString[kMaxLen + 1] = { 0 };
-        status = ConfigurationMgr().GetSoftwareVersionString(softwareVersionString, sizeof(softwareVersionString));
+        status = GetDeviceInstanceInfoProvider()->GetSoftwareVersionString(softwareVersionString, sizeof(softwareVersionString));
```

Comment:
Will this break API consumers who were configuring the software version string via the configuration manager and using the generic device instance info provider?  Or does no one use the generic thing out-of-tree so the new required virtual method will get noticed?

Reply:
This change is unnecessary and confusing as it only moves software version string to `DeviceInstanceInfoProvider` while leaving software version number in `ConfigurationManager`. You could add `StoreSoftwareVersionString` to `ConfigurationManager` to achieve what you need without breaking existing code.

Reply:
+1 to just adding a `StoreSoftwareVersionString` *only* to POSIX ConfigurationManager (and NO OTHER)